dotCMS/src/main/java/com/dotcms/rest/api/v1/osgi/OSGIResource.java

```diff
@@ -495,7 +495,10 @@ public final ResponseEntityStringView modifyExtraPackages(@Context final HttpSer
         final String extraPackages = extraPackagesForm.getPackages();
 
         Logger.debug(this, ()->"Modifying the extra packages OSGI jar " + extraPackages);
-        OSGIUtil.getInstance().writeOsgiExtras(extraPackages);
+
+        final boolean testDryRun = request.getParameter("testDryRun") != null?
+                Boolean.parseBoolean(request.getParameter("testDryRun")): true;
```

Comment:
As picked up by SonarQube,  you do not need the separate null check Boolean.parseBoolean is null safe internally it does "true".equalsIgnoreCase(s)  which will just return false if s is null. 

Fix the other new issues introduced also other than the use of deprecated PACKAGE_SPECIFICATION_VERSION if we need this.